### DISCUSSION TITLE
Gradient logo

### DISCUSSION THREAD

**Discussion by omnigenous** [fastfetch-cli/fastfetch#963] [Q&A] [answered]
I have tried:

`fastfetch --logo arch --logo-type small --logo-color-1 red`

But adding `--logo-color-[2-9]` doesn't not change anything. At least with small arch logo version.

Is it possible to apply `lolcat` to _logo only_?

**Comment by CarterLi** [comment]
`fastfetch -l arch -c none -s none | lolcat --force | fastfetch --raw - --logo-width 36 --logo-height 20`

I suggest you not do this.

**Reply by omnigenous** [reply]
Thank you for response! In my case I had to modify it to:
`fastfetch -l arch -s none | lolcat --force | fastfetch --raw - --logo-width 14 --logo-height 8`
because I do want _small_ logo specifically.
The only problem is there two empty lines at the top and bottom.  When I tried piping everything to `head`, `tail` or `sed` it ether breaks output or removes colors from modules but not from logo.

I want to have 1 empty line/space at the top and bottom, here is my config:

<details>

```json
{
  "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
  "logo": {
    "type": "small",
    "padding": {
      "top": 1,
      "left": 1
    },
    "color": {
      "1": "yellow"
    }
  },
  "display": {
    "color": "yellow"
  },
  "modules": [
    "break",
    {
      "type": "custom",
      "key": "key",
      "format": "example1"
    },
    {
      "type": "custom",
      "format": "example2"
    },
    {
      "type": "custom",
      "format": "example3"
    },
    {
      "type": "custom",
      "format": "example4"
    },
    {
      "type": "custom",
      "format": "example5"
    },
    {
      "type": "custom",
      "format": "example6"
    },
    {
      "type": "custom",
      "format": "example7"
    },
    "break"
  ]
}
```
</details>

![2024-05-24_10-15-17](https://github.com/fastfetch-cli/fastfetch/assets/154441415/ed91a323-593d-4932-aabb-7d16399ece3a)
![2024-05-24_10-15-31](https://github.com/fastfetch-cli/fastfetch/assets/154441415/92917f36-00ff-4630-a9e4-ccac9030fc5c)

**Comment by CarterLi** [comment]
> But adding `--logo-color-[2-9]` doesn't not change anything. At least with small arch logo version.

There must be color placeholders `%N` set in the logo source. eg. https://github.com/fastfetch-cli/fastfetch/blob/dev/src/logo/ascii/arch.txt

```
                  -`
                 .o+`
                `ooo/
               `+oooo:
              `+oooooo:
              -+oooooo+:
            `/:-:++oooo+:
           `/++++/+++++++:
          `/++++++++++++++:
         `/+++o$2oooooooo$1oooo/`
        ./$2ooosssso++osssssso$1+`
$2       .oossssso-````/ossssss+`
      -osssssso.      :ssssssso.
     :osssssss/        osssso+++.
    /ossssssss/        +ssssooo/-
  `/ossssso+/:-        -:/+osssso+-
 `+sso+:-`                 `.-/+oso:
`++:.                           `-/+/
.`                                 `/
```

`$N` means texts will be renderred in color of `--logo-color-N` starting there.

The small variant https://github.com/fastfetch-cli/fastfetch/blob/dev/src/logo/ascii/arch_small.txt doesn't have `$N`s

**Comment by omnigenous** [comment, marked as answer]
Ended up creating custom logo, see https://github.com/fastfetch-cli/fastfetch/discussions/964#discussioncomment-9546407

`cat custom_logo.txt | lolcat --force | fastfetch --raw - --logo-width 15 --logo-height 7 --logo-padding-left 0`

custom_logo.txt:
```
       /\
      /  \
     /    \
    /      \
   /   ,,   \
  /   |  |   \
 /_-''    ''-_\
```
config:
```
{
  "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
  "logo": {
    "type": "small",
    "padding": {
      "top": 1,
      "left": 1
    },
    "color": {
      "1": "yellow"
    }
  },
  "display": {
    "color": "yellow"
  },
  "modules": [
    "break",
    {
      "type": "custom",
      "key": "key",
      "format": "example1"
    },
    {
      "type": "custom",
      "format": "example2"
    },
    {
      "type": "custom",
      "format": "example3"
    },
    {
      "type": "custom",
      "format": "example4"
    },
    {
      "type": "custom",
      "format": "example5"
    },
    {
      "type": "custom",
      "format": "example6"
    },
    {
      "type": "custom",
      "format": "example7"
    },
    "break"
  ]
}
```

**Reply by omnigenous** [reply]
@CarterLi Hello. I think this is no longer works after last update.

```
Logo: Failed to load file content from logo source: -
```

Removing lolcat helps but then I lose gradient 😥

**Reply by CarterLi** [reply]
Works for me

<img width="2540" height="1375" alt="image" src="https://github.com/user-attachments/assets/d7f34cca-459f-4c73-9740-27a5a1ac1363" />
